src/main/scala/mesosphere/marathon/api/v2/LeaderResource.scala

```diff
@@ -49,14 +54,19 @@ class LeaderResource @Inject() (
   def delete(
     @QueryParam("backup") backupNullable: String,
     @QueryParam("restore") restoreNullable: String,
-    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
-    withAuthorization(UpdateResource, AuthorizedResource.Leader) {
+    @Context req: HttpServletRequest,
+    @Suspended asyncResponse: AsyncResponse): Unit = sendResponse(asyncResponse) {
+    async {
+      implicit val identity = await(authenticatedAsync(req))
+      checkAuthorization(UpdateResource, AuthorizedResource.Leader)
```

Comment:
why not `withAuthorization`?

Reply:
Async/await cannot be nested in functions and `withAuthorization` takes a closure. I couldn't find a different way.